src/lucky/text_response.cr

```diff
@@ -1,4 +1,13 @@
+{% if !flag?(:without_zlib) %}
+  require "gzip"
+{% end %}
+
 class Lucky::TextResponse < Lucky::Response
+  Habitat.create do
+    setting gzip_enabled : Bool = false
+    setting gzip_content_types : Array(String) = %w(text/html text/javascript application/json text/plain application/xml text/csv)
```

Comment:
I'm thinking these settings might go perfectly in https://github.com/luckyframework/lucky/blob/master/src/lucky/server.cr#L6 with some of the other settings.

Then my second suggestion is to share the `gzip_enabled` setting with the `StaticCompressionHandler` in https://github.com/luckyframework/lucky_cli/pull/442/files rather than having 2 different settings. I assume if people turn it off for one it is because they either don't want gzip for anything, or they have nginx or something else that is compressing both responses.

If enough people want them to be separate we can separate them later, but I think trying to have on setting would be great if we can get away with it

Reply:
Yep, this sounds good. I'll also change #982 to use this setting.

I'll also expand the content types a little and use them for the static handler and text response. The static handler is currently still using file extensions.

Reply:
Sounds great!

Reply:
Is it ok to merge my changes for #982 into this PR? That would make it a little easier to deal with the shared settings as well as working with Lucky Cli.

Reply:
Yep! I think that's a good idea